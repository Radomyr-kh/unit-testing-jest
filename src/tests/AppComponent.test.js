import App from '../components/App/index';
import renderer from 'react-test-renderer';

describe('Tests for App Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = renderer.create(<App />);
  });

  it('renders correctly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders H1 element with 'Hello World' ", () => {
    const root = wrapper.root;
    expect(root.findByType('h1')).toBeTruthy();
  });
});
