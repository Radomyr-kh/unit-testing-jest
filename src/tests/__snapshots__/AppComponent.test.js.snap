// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App Component renders correctly 1`] = `
<div
  className="App"
>
  <h1>
    Hello World!
  </h1>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul>
      <li>
        error
        : 
        request error
      </li>
    </ul>
  </div>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul>
      <li>
        error
        : 
        request error
      </li>
    </ul>
  </div>
</div>
`;
